core/src/test/java/org/pentaho/di/connections/ConnectionManagerTest.java

```diff
@@ -61,13 +72,56 @@ public class ConnectionManagerTest {
 
   private ConnectionManager connectionManager;
 
-  private MemoryMetaStore memoryMetaStore = new MemoryMetaStore();
+  private Bowl bowl;
+
+  private MemoryMetaStore memoryMetaStore;
+
+
+  @BeforeClass
+  public static void setupClass() throws Exception {
+    KettleClientEnvironment.init();
+    DefaultBowl defaultBowl = DefaultBowl.getInstance();
+    assertNotNull( defaultBowl );
+
+    // Make sure that the default bowl has a meta-store, as this is a requirement of the connection manager,
+    // and otherwise would affect the default connection manager.
+    IMetaStore defaultMemoryMetaStore = new MemoryMetaStore();
+    defaultBowl.setMetastoreSupplier( () -> defaultMemoryMetaStore );
+  }
 
   @Before
   public void setup() throws Exception {
-    KettleClientEnvironment.init();
-    connectionManager = new ConnectionManager();
-    connectionManager.setMetastoreSupplier( () -> memoryMetaStore );
+    bowl = mock( Bowl.class );
+    memoryMetaStore = new MemoryMetaStore();
+    connectionManager = new ConnectionManager( () -> memoryMetaStore, bowl );
+  }
+
+  @Test
+  public void testGetInstanceHasDefaultBowlAndMetaStore() {
+    DefaultBowl defaultBowl = DefaultBowl.getInstance();
+    assertNotNull( defaultBowl );
+
+    ConnectionManager defaultConnectionManager = ConnectionManager.getInstance();
+    assertNotNull( defaultConnectionManager );
+
+    assertSame( defaultBowl, defaultConnectionManager.getBowl() );
+
+    // Testing that the value is DefaultBowl.getInstance().getMetastore() wouldn't work due to tests running in
+    // parallel. The actual meta-store captured by the default connection manager instance can vary due to multiple
+    // threads calling the @BeforeClass method.
+    assertNotNull( defaultConnectionManager.getMetastoreSupplier().get() );
+  }
+
+  @Test
+  public void testGetInstanceOfBowlRespectsGivenBowlAndMetaStore() throws MetaStoreException {
+    IMetaStore metaStore = mock( IMetaStore.class );
+    Bowl bowl = mock( Bowl.class );
+    when( bowl.getMetastore() ).thenReturn( metaStore );
```

Comment:
I don't think this is used in this method. Nothing in this method should be using this metastore.